### ISSUE TITLE
update bmfs/forms bmfs/dimensions elliptic_curves/nfcurves on cloud

### ISSUE THREAD

**Issue by JohnCremona** [LMFDB/lmfdb-inventory#73]
I have started doing this, so only opening the Issue for the record.  I'll close it when done.

**Comment by JohnCremona** [comment]
There was a problem i updating nfcurves + rand and stats.  After creating the .new versions:
`
Renaming nfcurves to nfcurves.old and nfcurves.new to nfcurves in cloud database elliptic_curves...
MongoDB shell version: 3.2.17
connecting to: elliptic_curves
2017-10-14T12:52:47.852+0000 E QUERY    [thread1] Error: Authentication failed. :
DB.prototype._authOrThrow@src/mongo/shell/db.js:1441:20
@(auth):6:1
@(auth):1:2

exception: login failed
MongoDB shell version: 3.2.17
connecting to: elliptic_curves
2017-10-14T12:52:47.915+0000 E QUERY    [thread1] Error: Authentication failed. :
DB.prototype._authOrThrow@src/mongo/shell/db.js:1441:20
@(auth):6:1
@(auth):1:2

exception: login failed
Renaming nfcurves.rand to nfcurves.rand.old and nfcurves.rand.new to nfcurves.rand in cloud database elliptic_curves...
MongoDB shell version: 3.2.17
connecting to: elliptic_curves
2017-10-14T12:52:47.975+0000 E QUERY    [thread1] Error: Authentication failed. :
DB.prototype._authOrThrow@src/mongo/shell/db.js:1441:20
@(auth):6:1
@(auth):1:2

exception: login failed
MongoDB shell version: 3.2.17
connecting to: elliptic_curves
2017-10-14T12:52:48.035+0000 E QUERY    [thread1] Error: Authentication failed. :
DB.prototype._authOrThrow@src/mongo/shell/db.js:1441:20
@(auth):6:1
@(auth):1:2

exception: login failed
Renaming nfcurves.stats to nfcurves.stats.old and nfcurves.stats.new to nfcurves.stats in cloud database elliptic_curves...
MongoDB shell version: 3.2.17
connecting to: elliptic_curves
2017-10-14T12:52:48.095+0000 E QUERY    [thread1] Error: Authentication failed. :
DB.prototype._authOrThrow@src/mongo/shell/db.js:1441:20
@(auth):6:1
@(auth):1:2

exception: login failed
MongoDB shell version: 3.2.17
connecting to: elliptic_curves
2017-10-14T12:52:48.157+0000 E QUERY    [thread1] Error: Authentication failed. :
DB.prototype._authOrThrow@src/mongo/shell/db.js:1441:20
@(auth):6:1
@(auth):1:2

exception: login failed
...update complete.
After you verify that the new collections look good, be sure to run cleanup-cloud-db.sh to drop the .old collections.
`
My passwords.txt is exactly the same as in ~drew.  What is the problem?

**Comment by AndrewVSutherland** [comment]
The problem is that you are using an old version of update-cloud-db.sh, you need to pull the latest version into your lmfdb-gce directory (the latest updates on github are dated Sep 11, 2017, but you are using scripts dated Aug 21, 2017.

I just did the renaming manually (so the .new versions are now current and the previously current versions are now .old).  You should check that everything looks OK, and if so, pull from lmfdb-gce to get the lastest version of cleanup-cloud.sh and then use it to remove the .old collections.

**Comment by JohnCremona** [comment]
Thanks for sorting this.  The reason I did not pull a new version of lmfdb-gce was a simple git mistake:  git status said that I was ahead of upstream/master by 6 commits, which is confusing!  I was behind upstream, but my local copy of upstream/master was out of date.  So I did not pull.  I should have instantly realised that the script needed updating after we made changes in authentication.  Sorry.

I'll wait before ding the bmfs collections until you report that the number field copying is done.  Meanwhile I'll check over the ecnfs.
